### DISCUSSION TITLE
Flashing Newt with esptool and/or vscode+arduino

### DISCUSSION THREAD

**Discussion by paletochen** [Phambili-Tech/Newt_Display#60] [Q&A]
Hi,
Is there a way to securely flash the Newt with other tools than the arduino-ide?

I tried to flash with esptool but the Newt didin't boot afterwards. Several commands that i tried below:

```zsh
esptool.py --chip ESP32-S2 write_flash -z 0x1000 Newt-1.1.4-RC1.bin
esptool.py write_flash 0x1000 Newt-1.1.4-RC1.bin
```

I also tried vscode with the arduino extension.
While i managed to flash the `Hello World` app from the examples folder, I couldn't flash `Newt.ino`

Maybe Newt.ino needs some extra flags?
Or maybe flashing from a different memory address?

Thanks for clarifying

P.S: Linux/Debian Platform

**Comment by paletochen** [comment]
I found a mention to `esptool` in the [Issues tab](https://github.com/Phambili-Tech/Newt_Display/issues/2#issue-1313830300)

I then saw there an esptool command which provided more binaries, for bootloader, partitions, etc.

I downloaded the [v0.0.1.1 version](https://github.com/Phambili-Tech/Newt_Display_Hardware/blob/main/v0.0.11.zip) to get those binaries and successfully flashed 1.1.4-RC1 version

`esptool.py --chip esp32s2 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size 4MB 0x1000 Newt_0_0_11.ino.bootloader.bin 0x8000 Newt_0_0_11.ino.partitions.bin 0xe000 boot_app0.bin 0x10000 Newt-1.1.4-RC1.bin`

After reboot, my WiFi settings were still there (why aren't they erased?) and the device successfully updated to 1.1.5

Is this then the way to flash with `esptool`?
If so, could you please have a link in this repo to the proper binary files, in case these ones (v0.0.1.1) are not the latest ones?

I am still curious to understand if flashing using vscode would be possible, using the `.ino` file.

**Comment by darianbjohnson** [comment]
I would think flashing via VSCode would work - though I've never tried it. And I'll make sure to make all the bin files (bootloader and partitions) available for the next release

**Comment by paletochen** [comment]
I guess it doesn't work because of the lack of those bin files. vscode doesn't know how to handle the flashing nor have the binaries to do it.
Let me know when you'll add them so i can test it in my device.

Thanks

**Comment by darianbjohnson** [comment]
Note - I haven't forgotten about this... I'm just a little swamped with the day job. I promise - it's on my todo list
